### PR TITLE
[612] Category & Phrase Analytics

### LINKED ISSUE #612: Mixpanel Integration: Custom Category/Phrase Analytics
The purpose of this ticket is to implement Mixpanel Integration so that we can track events related to Custom Categories. 

**Acceptance Criteria:** 
- All events tracked are not tied to user private information
- Events track meets apple requirements to maintain [App Store Privacy disclosure](https://developer.apple.com/app-store/app-privacy-details/#linked-data) of "Data Not Linked to You"

**Events captured includes:**
1. New Custom Category Created
2. New Phrase Created from settings
3. New phrase created from Add Phrase
4. Preset Category Deletion (Personal Care, General, Basic Needs etc.) 
5. ~~Custom Category Deletion~~ (Team decided to not track this event.)
6. Preset Category Renaming (clicked into, changed and saved)
7. ~~Custom Category Renaming~~ (clicked into, changed and saved) (Team decided to not track this event.)
8. Preset "Show Category" turned ~~on and~~ off
9. ~~Custom Category "Show Category" turned on and off~~ (Team decided to not track this event.)

Links:

- [Tracking Plan Provided by Analytics Team] - See WT Confluence

### PR DESCRIPTION
**Pull request by jsmorgan42**
closes #612

# Description of Work
This PR adds Mixpanel events which fire when adding, creating, deleting, and renaming categories & phrases.

## Notes to Test
- Some of these events did not have exact specs on the spec sheet which may need to be ironed out on what the exact requirements should be for each event

---

### CODE REVIEW

File: Vocable/Analytics/Analytics.swift [hunk 1]
```diff
@@ -143,19 +144,43 @@ extension Analytics.Event {
...
+        Self(name: "Category Hidden", properties: ["Category": category.analyticsName])
+    }
+
+    static func phraseFavorited() -> Self {
+        Self(name: "Phrase Favorited")
+    }
+
+    static func phraseSaved() -> Self {
+        Self(name: "Phrase Saved")
+    }
```
Comment (moyerr) [inline Vocable/Analytics/Analytics.swift]: All these without parameters could just be `static var`s

File: Vocable/Extensions/Category+Helpers.swift [hunk 2]
```diff
@@ -66,6 +66,8 @@ extension Category {
         return true
     }
 
+    var analyticsName: String { isUserGenerated ? "Custom Category" : name ?? "" }
```
Comment (moyerr) [inline Vocable/Extensions/Category+Helpers.swift]: Based on this, it looks like `customCategoryAdded()` above could be using `analyticsName` just like the other Events.

File: Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift [hunk 3]
```diff
@@ -280,6 +280,9 @@ final class EditCategoryDetailViewController: VocableCollectionViewController {
             }
             category.isHidden = newHiddenState
             do {
+                if newHiddenState == true {
+                    Analytics.shared.track(.categoryHidden(category))
+                }
```
Comment (Clstroud) [inline Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift]: 1. Should this be restricted to preset categories?
2. Should it only be reported when the user has not renamed it? (as with the Phrase event we discussed)

File: Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift [hunk 4]
```diff
@@ -344,6 +347,7 @@ final class EditCategoryDetailViewController: VocableCollectionViewController {
             }
 
             do {
+                Analytics.shared.track(.categoryRemoved(category))
```
Comment (Clstroud) [inline Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift]: Same question here as ^^ with respect to preset restriction

File: Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift (hunk 3 above)
Comment (jsmorgan42) [reply Vocable/Features/Settings/EditCategories/EditCategoryDetailViewController.swift]: Those are some fair questions, probably would be good to sync up with Kyli to figure out our intent with these analytics events.

File: Vocable/Analytics/Analytics.swift [hunk 5]
```diff
@@ -143,4 +143,26 @@ extension Analytics.Event {
...
+        Self(name: "Preset Category Removed", properties: ["Category": category.name])
+    }
+
+    static func presetCategoryEdited(_ category: Category) -> Self {
+        Self(name: "Preset Category Edited", properties: ["Category": category.name])
+    }
+
+    static func categoryHidden(_ category: Category) -> Self {
+        Self(name: "Preset Category Hidden", properties: ["Category": category.name])
+    }
```
Comment (Clstroud) [inline Vocable/Analytics/Analytics.swift]: Should these be renamed to all specify "presetCategory..." for accuracy/uniformity?

Comment (jsmorgan42) [reply Vocable/Analytics/Analytics.swift]: That's probably a good idea

**Review by Clstroud — APPROVED** [review]
After existing comment is updated I think this will be in good shape. Nice job 👍🏻 🏆

**Comment by CananArikan** [comment]
Test verified. PR captures the events that are written in the AC. 
Test Device: iPhone 11 Simulator, Version 1.4.4-1

**Pull request merged**